dlt/common/schema/schema.py

```diff
@@ -780,11 +781,19 @@ def set_schema_contract(self, settings: TSchemaContract) -> None:
     def _infer_column(
         self, k: str, v: Any, data_type: TDataType = None, is_variant: bool = False
     ) -> TColumnSchema:
-        column_schema = TColumnSchema(
-            name=k,
-            data_type=data_type or self._infer_column_type(v, k),
-            nullable=not self._infer_hint("not_null", k),
-        )
+        # return unbounded table
+        if v is None and data_type is None:
+            column_schema = TColumnSchema(
+                name=k,
+                nullable=True,
```

Comment:
IMO this should be `not self._infer_hint("not_null", k)` as well. and if it is `False` then raise `raise CannotCoerceNullException(self.name, table_name, col_name)`

Reply:
also unit test this in common schema tests